xds/src/main/java/io/grpc/xds/internal/sds/SdsProtocolNegotiators.java

```diff
@@ -370,4 +361,42 @@ public void onException(Throwable throwable) {
       );
     }
   }
+
+  public static class FilterChainMatchingNegotiatorServerFactory
```

Comment:
I think this was discussed elsewhere: why is this class defined in `SdsProtocolNegotiators`? Ideally this should be its own top level class in `io.grpc.xds` .